File: Python_Products_Api/models/InputLine.py
class InputLine:

    def __init__(self, input_line: list[str]):
        self.__product_id = input_line[0]
        self.__recommendation_id = input_line[1]
        self.__rank = float(input_line[2])

    @property
    def product_id(self):
        return self.__product_id

    @property
    def recommendation_id(self):
        return self.__recommendation_id

    @property
    def rank(self):
        return self.__rank

File: MVideo/c#/TestMVideo/Models/InputLine.cs
#region using

using System.Globalization;

#endregion

namespace TestMVideo.Models
{
    public struct InputLine
    {
        public InputLine(string csvLine)
        {
            var parameters = csvLine.Split(',');
            ProductId = parameters[0];
            RecommendationId = parameters[1];
            Rank = float.Parse(parameters[2], NumberStyles.Any, CultureInfo.InvariantCulture);
        }

        public string ProductId { get; set; }
        public string RecommendationId { get; set; }
        public float Rank { get; set; }
    }
}